From 1e355df093ba930261d3b46bb711510576a48b42 Mon Sep 17 00:00:00 2001
From: "sungmok.shin" <sungmok.shin@lge.com>
Date: Tue, 30 Apr 2024 14:40:38 +0900
Subject: [PATCH] Use human readable process name for chrome inspector

Release Notes:
Use human readable process name for chrome inspector

:Detailed Notes:
This commit patches nodejs to display the human readable process
name at the target title in the chrome inspector. Chrome inspector used
to show duplicate title filled with script name when multiple targets
are running in the remote target.

A patch was merged to fix the duplicate client name. However, we faced
the problem that garbage character is appended after the target title.
Investigating further, it has been found out that getTargetTitle returns
human readable name when script name is empty. Since process name is
useful for developer to distinguish the remote target, let us patch
getTargetTitle to return only human readable process name.

MJ: Simplify after upstream change:
https://github.com/nodejs/node/pull/31633

:Testing Performed:

:QA Notes:

:Issues Addressed:
[PLAT-97316] Contains unnecessary characters in title of node inspect
---
Upstream-Status: Pending

 src/inspector_io.cc  | 6 +++++-
 src/node_internals.h | 1 +
 src/util.cc          | 4 ++++
 3 files changed, 10 insertions(+), 1 deletion(-)

diff --git a/src/inspector_io.cc b/src/inspector_io.cc
index 4e76e529fcf..e732a65f2d3 100644
--- a/src/inspector_io.cc
+++ b/src/inspector_io.cc
@@ -362,7 +362,11 @@ std::vector<std::string> InspectorIoDelegate::GetTargetIds() {
 }
 
 std::string InspectorIoDelegate::GetTargetTitle(const std::string& id) {
-  return script_name_.empty() ? GetHumanReadableProcessName() : script_name_;
+  /*
+   * Return human readable process name instead of script name for webOS
+   * Refer to PLAT-97316 for more detail
+   */
+  return GetHumanReadableProcessNameWithoutPID();
 }
 
 std::string InspectorIoDelegate::GetTargetUrl(const std::string& id) {
diff --git a/src/node_internals.h b/src/node_internals.h
index 85b666e11f5..5a4bf792611 100644
--- a/src/node_internals.h
+++ b/src/node_internals.h
@@ -109,6 +109,7 @@ void SignalExit(int signal, siginfo_t* info, void* ucontext);
 
 std::string GetProcessTitle(const char* default_title);
 std::string GetHumanReadableProcessName();
+std::string GetHumanReadableProcessNameWithoutPID();
 
 v8::Maybe<void> InitializeBaseContextForSnapshot(
     v8::Local<v8::Context> context);
diff --git a/src/util.cc b/src/util.cc
index 34263cc9f3d..1265e00d9ef 100644
--- a/src/util.cc
+++ b/src/util.cc
@@ -198,6 +198,10 @@ std::string GetHumanReadableProcessName() {
   return SPrintF("%s[%d]", GetProcessTitle("Node.js"), uv_os_getpid());
 }
 
+std::string GetHumanReadableProcessNameWithoutPID() {
+  return SPrintF("%s", GetProcessTitle("Node.js"));
+}
+
 std::vector<std::string_view> SplitString(const std::string_view in,
                                           const std::string_view delim) {
   std::vector<std::string_view> out;
